test/integration/lib/azul_search_service_test.rb

```diff
@@ -28,57 +24,129 @@ class AzulSearchServiceTest < ActiveSupport::TestCase
                                name_prefix: 'Azul Search Test',
                                user: @user,
                                test_array: @@studies_to_clean)
-    SearchFacet.update_all_facet_filters
+    @mock_facet_query = {
+      genusSpecies: {
+        is: ['Homo sapiens']
+      }
+    }.with_indifferent_access
+    @mock_term_query = {
+      sampleDisease: {
+        is: ['chronic obstructive pulmonary disease', 'idiopathic pulmonary fibrosis', 'pulmonary fibrosis']
+      }
+    }.with_indifferent_access
+    @terms_to_facets = [
+      {
+        id: 'disease',
+        filters: [
+          { id: 'chronic obstructive pulmonary disease', name: 'chronic obstructive pulmonary disease' },
+          { id: 'idiopathic pulmonary fibrosis', name: 'idiopathic pulmonary fibrosis' },
+          { id: 'pulmonary fibrosis', name: 'pulmonary fibrosis' }
+        ]
+      }.with_indifferent_access
+    ]
+    tcell_json = File.open(Rails.root.join('test/test_data/azul/human_tcell.json')).read
+    thymus_json = File.open(Rails.root.join('test/test_data/azul/human_thymus.json')).read
+    fibrosis_json = File.open(Rails.root.join('test/test_data/azul/pulmonary_fibrosis.json')).read
+    @human_tcell_response = JSON.parse(tcell_json).with_indifferent_access
+    @human_thymus_response = JSON.parse(thymus_json).with_indifferent_access
+    @fibrosis_response = JSON.parse(fibrosis_json).with_indifferent_access
   end
 
-  after(:all) do
-    SearchFacet.all.map(&:update_filter_values!)
+  test 'should search Azul using facets' do
+    mock = MiniTest::Mock.new
+    mock.expect :format_query_from_facets, @mock_facet_query, [@facets]
+    mock.expect :merge_query_objects, @mock_facet_query, [@mock_facet_query, nil]
+    mock.expect :projects, @human_tcell_response, [{ query: @mock_facet_query }]
+    ApplicationController.stub :hca_azul_client, mock do
+      results = AzulSearchService.get_results(selected_facets: @facets, terms: nil)
+      mock.verify
+      assert_includes results.keys, @hca_project_shortname
+      project = results[@hca_project_shortname]
+      # will always be project manifest file
+      manifest = project[:file_information].detect { |f| f[:file_type] == 'Project Manifest' }
+      assert manifest.present?
+    end
   end
 
-  test 'should search Azul using facets' do
-    results = AzulSearchService.get_results(selected_facets: @facets, terms: nil)
-    assert_includes results.keys, @hca_project_shortname
-    project = results[@hca_project_shortname]
-    # will always be project manifest file
-    manifest = project[:file_information].detect { |f| f[:file_type] == 'Project Manifest' }
-    assert manifest.present?
+  test 'should search Azul using numeric facets' do
+    facets = [
+      {
+        id: 'organism_age', filters: { min: 1, max: 5, unit: 'years' }
+      }.with_indifferent_access
```

Comment:
Is there any more elegant way to enable JSON-like format?  Needing `.with_indifferent_access` on every declaration is noisy.

Reply:
Totally agreed - this is my major gripe about Ruby, Hash objects, and Symbols.  Because `{ foo: 'bar' } != { 'foo' => 'bar' }`, we need  `with_indifferent_access`.  Enabling this as the default project-wide would require monkey patching the `Hash` class, and replacing it with `ActiveSupport::HashWithIndifferentAccess`.  I have no idea what the downstream consequences of this would be, especially with 3rd party libraries, so I've always resigned myself to needing to use `with_indifferent_access` all over the place.

But I agree - it's a pain, especially when I forget, and tests start failing all over the place.

Reply:
double concur.  IIRC the author of Ruby has said the handling of symbols and hashes is one of his biggest regrets in the design of the language.   And yeah, I don't know of any workaround that isn't even worse